### PR TITLE
DFR-907 Added a new method to remove deleted files from case data. Also add n…

### PR DESCRIPTION
**Pull request by FaisalMoJ**
…ew classes and controller

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFR-997

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ManageCaseDocumentsController.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.UploadCaseFilesAboutToSubmitHandler;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.AUTHORIZATION_HEADER;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/case-orchestration")
+public class ManageCaseDocumentsController extends BaseController {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ManageCaseDocumentsController.java:27]: Please migrate this controller to a handler as per dfr1018

**Review by gmikerlemos — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ManageCaseDocumentsController.java (hunk 1 above)
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ManageCaseDocumentsController.java:27]: Hi Mike, thanks yah currently working on it to migrate to the generic controller as per DFR-1018 but i had a chat with Ruban and we are of the opinion the PR could be reviewed while we work on migrating it to the new way of controller work

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadCaseFilesAboutToSubmitHandlerTest.java [hunk 2]
```diff
@@ -595,6 +599,72 @@ public void givenNoUploadFileWhenAboutToSubmitThenNoErrors() {
...
+    @Test
+    public void removeDeletedFileFromCaseDataCollection() {
+
+        uploadCaseFilesService.removeDeletedFilesFromCaseData(populateCaseData().getData());
+
+        assertThat(getDocumentCollection(caseData, RESP_CHRONOLOGIES_STATEMENTS_COLLECTION), hasSize(1));
+    }
+
+    @Test
+    public void applicantAndRespondentKeysDoNotExistInCaseData() {
```
Comment (gmikerlemos) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadCaseFilesAboutToSubmitHandlerTest.java:637]: please use the given_When_Then test name structure

**Review by gmikerlemos — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToStartCaseHandler.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.UploadCaseFilesAboutToSubmitHandler;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ManageCaseDocumentsAboutToStartCaseHandler implements CallbackHandler {
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToStartCaseHandler.java:16]: Renamed class name as suggested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToSubmitCaseHandler.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.UploadCaseFilesAboutToSubmitHandler;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ManageCaseDocumentsAboutToSubmitCaseHandler implements CallbackHandler {
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToSubmitCaseHandler.java:16]: Renamed class name as suggested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToStartCaseHandler.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.MANAGE_CASE_DOCUMENTS.equals(eventType);
+    }
+
+    @Override
+    public AboutToStartOrSubmitCallbackResponse handle(CallbackRequest callbackRequest, String userAuthorisation) {
+        return uploadCaseFilesAboutToSubmitHandler.handle(callbackRequest.getCaseDetails().getData());
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToStartCaseHandler.java]: Refactored handler class in service as suggested by Mike

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToSubmitCaseHandler.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.MANAGE_CASE_DOCUMENTS.equals(eventType);
+    }
+
+    @Override
+    public AboutToStartOrSubmitCallbackResponse handle(CallbackRequest callbackRequest, String userAuthorisation) {
+        return uploadCaseFilesAboutToSubmitHandler.handle(uploadCaseFilesAboutToSubmitHandler.removeDeletedFilesFromCaseData(
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManageCaseDocumentsAboutToSubmitCaseHandler.java]: Same as above refactored the code in service class

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/DocumentDetailsData.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class DocumentDetailsData {
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/DocumentDetailsData.java:15]: This is the convention used in COS service for our model classes so i followed that as well 🤔

[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/DocumentDetailsData.java:15]: Renamed the collection class used in data to the name as Mike suggested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadCaseFilesAboutToSubmitHandler.java [hunk 8]
```diff
@@ -586,4 +596,102 @@ private boolean isTrialBundle(ContestedUploadedDocument uploadedCaseDocument) {
...
+                allDocuments.addAll(getDocumentCollection(caseData, collection));
+            }
+        }
+
+        return allDocuments;
+    }
+
+    private Set<String> getContestedUploadDocumentCollections() {
+
+        return Set.of(APPLICANT_CORRESPONDENCE_COLLECTION, APPLICANT_FR_FORM_COLLECTION, APPLICANT_EVIDENCE_COLLECTION,
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadCaseFilesAboutToSubmitHandler.java]: Changed it to enum

**Comment by FaisalMoJ** [comment]
Closing this PR as this is now tracked by PR #723

**Pull request closed without merging**